test/inductor/test_cpu_repro.py

```diff
@@ -2878,6 +2879,35 @@ def forward(self, x):
         self.assertRaises(RuntimeError, lambda: func(example_inputs))
         self.assertRaises(RuntimeError, lambda: jit_func(example_inputs))
 
+    def test_nn_param_assign(self):
+        # https://github.com/pytorch/pytorch/issues/99569
+        class Model2(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv = nn.Conv2d(in_channels=3, out_channels=5, kernel_size=3)
+                self.batchnorm = nn.BatchNorm2d(num_features=5)
+                self.conv_weight = torch.randn(5, 3, 3, 3)
+                self.conv_bias = torch.randn(5)
+
+            def forward(self, x):
+                self.conv.weight = nn.Parameter(self.conv_weight)
+                self.conv.bias = nn.Parameter(self.conv_bias, requires_grad=False)
+                self.conv.eval()
```

Comment:
This is not cool haha
It should be a no-op though so not sure it changes much? Since you don't test it explicitly, I would remove it.

Unless you meant to call this on the batchnorm where it will change its behavior!

Reply:
This is just copied from the repro provided in https://github.com/pytorch/pytorch/issues/99569

I agree it doesn't make much sense, but want to make sure we don't break on that user report.